Sources/Fuzzilli/Engines/FuzzEngine.swift

```diff
@@ -20,15 +20,68 @@ public protocol FuzzEngine: ComponentBase {
 }
 
 extension FuzzEngine {
+    private func instrumentForDifferentialTesting(_ program: Program) -> Program {
+        let expectedProbed = Int(fuzzer.config.differentialRate * Double(program.code.count))
+        let b = fuzzer.makeBuilder()
+        var reassignVars: [Variable] = []
+        for _ in 0..<expectedProbed {
+            reassignVars.append(b.loadInt(0))
+        }
+
+        var reassignedVars = 0
+        let differentialHashContext = DifferentialHash().requiredContext
+        b.adopting(from: program) {
+            for instr in program.code {
+                b.adopt(instr)
+
+                guard differentialHashContext.isSubset(of: b.context) else { continue }
+                guard b.numVisibleVariables > reassignVars.count else { continue }
+
+                if probability(fuzzer.config.differentialWeaveRate) {
+                    for _ in 1...10 {
+                        let v = b.randVar()
+                        if !reassignVars.contains(v) {
+                            b.differentialHash(v)
+                            break
+                        }
+                    }
+                }
+
+                guard reassignedVars < reassignVars.count else { continue }
+                if probability(fuzzer.config.differentialRate) {
+                    b.reassign(b.randVar(), to: reassignVars[reassignedVars])
+                    reassignedVars += 1
+                }
+            }
+        }
+
+        for v in reassignVars {
+            b.differentialHash(v)
+        }
+
+        return Program(code: b.finalize().code, parent: program.parent,
+                       comments: program.comments, contributors: program.contributors)
+    }
+
     public func execute(_ program: Program) -> ExecutionOutcome {
         fuzzer.dispatchEvent(fuzzer.events.ProgramGenerated, data: program)
 
-        let execution = fuzzer.execute(program)
+        let postprocessedProgram: Program
+        if fuzzer.config.differentialRate > 0.0 || fuzzer.config.differentialWeaveRate > 0.0 {
```

Comment:
Nit: maybe add a single `fuzzer.config.performDifferentialTesting` or so member (can be computed and return this expression here)?